### ISSUE TITLE
[Datasource Loki] Dynamic value for "log-range" in count_over_time function

### ISSUE THREAD

**Issue by sherifkayad** [grafana/grafana#35100]
**What would you like to be added**:
Currently it's not possible to use the selected Grafana time range (e.g. relative last 5m, 1h, .. 3d, .. etc. nor absolute from-to) in querying Loki with the function "count_over_time" e.g. using the pre-defined variable `$__range` or `$__timeFilter`. Currently only Prometheus data sources can use it as per https://grafana.com/docs/grafana/latest/variables/variable-types/global-variables/#__range

I would love to have one of those variables that can be passed to the `count_over_time` function so that dashboards using Loki as datasource can query Loki functions and visualize that without having to either hardcode relative times or provide that as an additional Dashboard Variable.

So basically my main issue is that the `count_over_time` function only accepts relative time ranges strings (e.g. 5m, 1h, 3h, .. etc.) and even those can't be referenced from a pre-defined Grafana variable.

**Why is this needed**:
Basically unleashing the power of querying Loki and visualizing that in Grafana Dashboards also for metrics related to logs / log entries and so on with full flexibility.

![image](https://user-images.githubusercontent.com/8362784/120474783-e7741a80-c3a8-11eb-8a4b-ca1e9006a17b.png)

![image](https://user-images.githubusercontent.com/8362784/120474908-0a063380-c3a9-11eb-9568-4b94c30e2f5d.png)
